TypeScript.YAML-tmLanguage

```diff
@@ -2536,11 +2537,30 @@ repository:
       match: '{{startOfIdentifier}}keyof{{endOfIdentifier}}'
     - name: keyword.operator.ternary.ts
       match: (\?|\:)
-    - name: keyword.operator.expression.infer.ts
-      match: '{{startOfIdentifier}}infer(?=\s+[_$[:alpha:]])'
     - name: keyword.operator.expression.import.ts
       match: '{{startOfIdentifier}}import(?=\s*\()'
 
+  type-infer:
+    patterns:
+    # try to match infer..extends on a single line
+    - match: '{{startOfIdentifier}}(infer)\s+({{identifier}})\s+(extends){{endOfIdentifier}}'
+      name: meta.type.infer.ts
+      captures:
+        '1': { name: keyword.operator.expression.infer.ts }
+        '2': { name: entity.name.type.ts }
+        '3': { name: keyword.operator.expression.extends.ts }
+    # alternatively, match across multiple lines
+    - begin: '{{startOfIdentifier}}infer(?=\s+[_$[:alpha:]])'
+      beginCaptures:
+        '0': { name: keyword.operator.expression.infer.ts }
+      end: '(?=[;),}\]:?\-\+\>''"`|&]|\|\||\&\&|\!\=\=)'
```

Comment:
Why do we need to match this across multiple lines. is it common if not i think we should just not add it..  
Concern here is the ending regex not containing line end or beginning or line in some form... On Single line matching it ends at extends so why not same? so why do we need to match more than extends.. 
We try to end things in type space with new line to ensure we aren't coloring things as `type` because of many ambiguity and textmate's requirement to look ahead only line at a time..

Reply:
Primarily because there's no NLT restriction between `infer T` and `extends`. I can drop the multi-line support if it doesn't seem worthwhile.

Reply:
i think its better to skip it..  We can always add it if many people are interested and we see it as more value